keyboards/aplusx_pulsar/xboard_qs/keymaps/via/keymap.c

```diff
@@ -0,0 +1,371 @@
+
+static bool isMacMode = false;
+
+static uint8_t Status_LED_Bright;       // maybe caps, scroll, PC1/2, win/mac, dynamic macro, win lock
+                                           // 0 = 초기값, 1~5 = 50~250의 5단계 밝기
+#define kvm_deadtime    100
+//------------------------------
+typedef union {
+    uint32_t raw;
+    struct {
+        bool    eeprom_kvm_pc_sel :1;
+        uint8_t eeprom_Status_LED_Bright;       //
+    };
+} user_config_t;
+
+user_config_t user_config;
+
+//-=-------------------------------------------
+    enum _layer { WIN_BASE = 0, WIN_FN, MAC_BASE, MAC_FN };
+
+    enum pulsar_keycodes {
+        KC_TGUI = QK_KB_0,  // Ver 0.21.6에서 변경 (USER00,)  // Toggle between GUI Lock or Unlock
+        KC_WIN_MODE,        // WINDOWS Keyboard로 변경
+        KC_MAC_MODE,        // Machintosh Keyboard로 변경
+        KC_TMODE,        // WINDOWS <--> Machintosh
+        KC_SPOTLIGHT,       // MAC_key
+        KC_DICTATION,       // MAC_key
+        KC_DO_NOT_DISTURB,  // MAC_key
+        KC_LOCK_SCREEN,     // MAC_key
+        KC_LOPTN,           // MAC_key - left option   (win위치)
+        KC_ROPTN,           // MAC_key - right option
+        KC_LCMMD,           // MAC_key - left command  (ALT위치)
+        KC_RCMMD,           // MAC_key - right command
+        KC_TASK_VIEW,       // for win (gui + tap)
+        KC_FILE_EXPLORER,   // for win (gui + e)
+        KC_PC1,             // KM Switch PC1
+        KC_PC2,             // KM Switch PC2
+        KC_TPC,             // KM Switch PC1<-->PC2 Toggle
+        INIT_3S,            // EE_CLR(EEPROM CLear)(공장초기화)를 3초이상 눌러야 동작
+        KC_S_LED,            // Change Status LED Brightness
+    };
+
+#define KC_WINM KC_WIN_MODE
+#define KC_MACM KC_MAC_MODE
+
+#define KC_SPLT KC_SPOTLIGHT
+#define KC_SIRI KC_DICTATION
+#define KC_DOND KC_DO_NOT_DISTURB
+#define KC_LOCK_ KC_LOCK_SCREEN
+
+#define KC_TASK LGUI(KC_TAB)
+#define KC_FLXP LGUI(KC_E)
+#define KC_DIC  LGUI(KC_H)      // WIN ; 받아쓰기
+#define KC_LOCK2 LGUI(KC_L)     // WIN ; pc잠금
+#define KC_SNAP SCMD(KC_5)      // MAC Capure ; commnd + shift + 5
+
+void kvm_switch(bool pc_num){
+    writePinHigh(GPIO_KM_OE);
+    writePinLow(GPIO_KM_PWEN);
+    if (!pc_num) writePinLow(GPIO_KM_SEL);
+    else   writePinHigh(GPIO_KM_SEL);
+    writePinLow(GPIO_KM_OE);
+
+    kvm_timer = timer_read();
+    kvm_sel_on = true;
+};
+
+void keyboard_pre_init_user(void) {
+    setPinOutput(GPIO_KM_OE);       // H/W power on default = 0
+    setPinOutput(GPIO_KM_SEL);      // H/W power on default = 0
+    setPinOutput(GPIO_KM_PWEN);     // H/W power on default = 1
+}
+
+void keyboard_post_init_user(void) {
+    user_config.raw = eeconfig_read_user();                 // Read the user config from EEPROM
+    kvm_pc_sel      = user_config.eeprom_kvm_pc_sel;
+    kvm_switch(kvm_pc_sel);
+
+    Status_LED_Bright = user_config.eeprom_Status_LED_Bright;   // 0 = 초기값, 1~5 = 50~250의 5단계 밝기
+    if (Status_LED_Bright == 0) Status_LED_Bright = 5;
+}
+
```

Comment:
This code should exist at the kb level, not the keymap.

Reply:
I learned a lot. Thank you